control-plane-operator/controllers/hostedcontrolplane/kas/params.go

```diff
@@ -289,12 +288,8 @@ func NewKubeAPIServerParams(ctx context.Context, hcp *hyperv1.HostedControlPlane
 		},
 	}
 
-	switch hcp.Spec.Platform.Type {
-	case hyperv1.AWSPlatform:
+	if hcp.Spec.Platform.Type == hyperv1.AWSPlatform {
 		params.CloudProvider = aws.Provider
-	case hyperv1.AzurePlatform:
-		params.CloudProvider = azure.Provider
```

Comment:
This doesn't need set for Azure - https://cloud-provider-azure.sigs.k8s.io/install/azure-ccm/#kube-apiserver